File: fuelguard-webclient/src/app/features/gemini-ai/models/gemini.models.ts
export interface GeminiChatRequestDto {
  question: string;
  stationHint?: string | null;
  periodHint?: string | null;
  additionalContext?: string | null;
}

export interface GeminiChatResponseDto {
  answer: string;
  riskLevel: string;
  recommendations: string[];
  explainability: string;
  poweredByGemini: boolean;
  disclaimer?: string | null;
}

export interface GeminiInsightsResponseDto {
  operationalInsights: string[];
  anomalyHighlights: string[];
  recommendations: string[];
  executiveSummary: string;
  riskObservations: string[];
  poweredByGemini: boolean;
  disclaimer?: string | null;
}

export interface GeminiChatMessage {
  id: string;
  role: 'user' | 'assistant';
  content: string;
  riskLevel?: string;
  recommendations?: string[];
  explainability?: string;
  poweredByGemini?: boolean;
  isTyping?: boolean;
}

File: src/FuelGuard.Application/Abstractions/Gemini/GeminiModels.cs
namespace FuelGuard.Application.Abstractions.Gemini;

public sealed record GeminiChatRequest(
    string Question,
    string? StationHint = null,
    string? PeriodHint = null,
    string? AdditionalContext = null);

public sealed record GeminiChatResponse(
    string Answer,
    string RiskLevel,
    IReadOnlyList<string> Recommendations,
    string Explainability,
    bool PoweredByGemini,
    string? Disclaimer = null);

public sealed record GeminiInsightsResponse(
    IReadOnlyList<string> OperationalInsights,
    IReadOnlyList<string> AnomalyHighlights,
    IReadOnlyList<string> Recommendations,
    string ExecutiveSummary,
    IReadOnlyList<string> RiskObservations,
    bool PoweredByGemini,
    string? Disclaimer = null);

public sealed record OperationalFactsSnapshot(
    string FactsBlock,
    string DominantRiskLevel,
    int AnomalyCount,
    int TransactionCount);
